actors/evm/src/interpreter/stack.rs

```diff
@@ -1,62 +1,97 @@
 #![allow(dead_code)]
 
-use {crate::interpreter::U256, arrayvec::ArrayVec, serde::Serialize};
+use crate::interpreter::U256;
 
 /// Ethereum Yellow Paper (9.1)
-pub const MAX_STACK_SIZE: usize = 1024;
+pub const STACK_SIZE: usize = 1024;
+
+const INITIAL_STACK_SIZE: usize = 32;
 
 /// EVM stack.
-#[derive(Clone, Debug, Default, Serialize)]
-pub struct Stack(pub ArrayVec<U256, MAX_STACK_SIZE>);
+#[derive(Clone, Debug)]
+pub struct Stack {
+    sk: Vec<U256>,
+    d: usize,
+}
 
 impl Stack {
     #[inline]
-    pub const fn new() -> Self {
-        Self(ArrayVec::new_const())
+    pub fn new() -> Self {
+        Stack { sk: Vec::from([U256::zero(); INITIAL_STACK_SIZE]), d: 0 }
```

Comment:
```suggestion
        Stack { sk: vec![U256::zero(); INITIAL_STACK_SIZE], d: 0 }
```

(avoids a copy)

Reply:
Alternatively, any reason not to write `Vec::with_capacity`?

Reply:
(avoids zeroing)

Reply:
Ah, initial.

Reply:
with capacity doesnt initialize, but fiven we do unchecked acces thats probably fine.

Reply:
Lets just avoid the copy for now.

Reply:
committed suggestion.